pkg/component/controller/extensions_controller.go

```diff
@@ -412,7 +412,14 @@ func (ec *ExtensionsController) leaseLost() {
 	ec.mux.Lock()
 	defer ec.mux.Unlock()
 	ec.L.Warn("Lost leader lease, stopping controller-manager")
-	ec.mgrCancelFn()
+
+	// We have no guarantees on concurrency here, so use mutex
+	ec.mux.Lock()
```

Comment:
That will deadlock :upside_down_face:

Reply:
oh, silly me. totally didn't see that we already had the lock there... 🤦